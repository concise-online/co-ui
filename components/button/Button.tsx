import React, { ButtonHTMLAttributes } from 'react';
import classNames from 'classnames';
import { getPrefixCls } from '../utils/config';

const prefixCls = getPrefixCls('btn');

interface Props extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
  type?: 'primary' | 'success' | 'info' | 'warning' | 'danger';
  htmlType?: 'button' | 'submit' | 'reset';
  fill?: boolean;
}

function Button(props: Props) {
  const { children, className, htmlType, type, fill, ...btnProps } = props;
  const classes = classNames(prefixCls, className, {
    [`${prefixCls}-${type}`]: !!type,
    [`${prefixCls}-fill`]: fill,
  });

  return (
    <button
      type={htmlType}
      className={classes}
      {...btnProps}
    >
      <span>{children}</span>
    </button>
  );
}

export default Button;
