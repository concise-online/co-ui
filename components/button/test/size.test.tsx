import React from 'react';
import { render, shallow } from 'enzyme';
import Button from '../index';

describe('button size feature', () => {
  it('render correctly', () => {
    const wrapper = render((
      <div>
        <Button size="large">Large</Button>
        <Button>Normal</Button>
        <Button size="small">Small</Button>
      </div>
    ));

    expect(wrapper).toMatchSnapshot();
  });

  it('button group size prop rewrite button size prop', () => {
    const wrapper = shallow((
      <Button.Group size="small">
        <Button size="small">Small</Button>
        <Button>Normal</Button>
        <Button size="large">Large</Button>
      </Button.Group>
    ));
    const children = wrapper.children();

    expect(children.someWhere((child) => child.prop('size') === 'small')).toBeTruthy();
  });
});
