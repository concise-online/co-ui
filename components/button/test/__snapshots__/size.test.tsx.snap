// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button size feature render correctly 1`] = `
<div>
  <button
    class="co-btn co-btn-large"
  >
    <span>
      Large
    </span>
  </button>
  <button
    class="co-btn"
  >
    <span>
      Normal
    </span>
  </button>
  <button
    class="co-btn co-btn-small"
  >
    <span>
      Small
    </span>
  </button>
</div>
`;
